src/qt/receivecoinsdialog.cpp

```diff
@@ -41,6 +43,17 @@ ReceiveCoinsDialog::ReceiveCoinsDialog(const PlatformStyle *_platformStyle, QWid
         ui->removeRequestButton->setIcon(_platformStyle->SingleColorIcon(":/icons/remove"));
     }
 
+    // configure bech32 checkbox, disable if launched with legacy as default:
+    if (model->getDefaultAddressType() == OUTPUT_TYPE_BECH32) {
+        ui->useBech32->setCheckState(Qt::Checked);
+    } else {
+        ui->useBech32->setCheckState(Qt::Unchecked);
+    }
+
+    if (model->getDefaultAddressType() == OUTPUT_TYPE_LEGACY) {
+        ui->useBech32->setEnabled(false);
```

Comment:
Might want to call setenabled unconditionally (`ui->useBech32->setEnabled(model->getDefaultAddressType() == OUTPUT_TYPE_LEGACY))`) in case we want to support changing default address type in the ui in the future.

Reply:
Done